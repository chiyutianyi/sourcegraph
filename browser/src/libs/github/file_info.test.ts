import { startCase } from 'lodash'
import { readFile } from 'mz/fs'
import { getFilePath } from './util'

describe('github/file_info', () => {
    for (const version of ['github.com', 'ghe-2.14.11']) {
        describe(version, () => {
            for (const extension of ['vanilla', 'refined-github']) {
                describe(startCase(extension), () => {
                    it('finds the file path', async () => {
                        document.body.innerHTML = await readFile(
                            `${__dirname}/__fixtures__/${version}/blob/${extension}/page.html`,
                            'utf-8'
                        )
                        expect(getFilePath()).toMatchSnapshot()
                    })
                })
            }
        })
    }
})
