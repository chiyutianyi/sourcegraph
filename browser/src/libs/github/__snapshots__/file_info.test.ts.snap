// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`github/file_info ghe-2.14.11 Refined Github finds the file path 1`] = `"bench_test.go"`;

exports[`github/file_info ghe-2.14.11 Vanilla finds the file path 1`] = `"bench_test.go"`;

exports[`github/file_info github.com Refined Github finds the file path 1`] = `"shared/src/api/extension/types/url.ts"`;

exports[`github/file_info github.com Vanilla finds the file path 1`] = `"shared/src/api/extension/types/url.ts"`;
